### DISCUSSION TITLE
Generic Event Types/Interfaces

### DISCUSSION THREAD

**Discussion by SortOmega** [primefaces/community#2246] [PrimeNG]
When coding, types and interfaces are also considered as documentation. It provides information about how the app works and also provides an easier way to code thanks to autocompletion when typing properties of objects.
One thing I have noticed when using the PrimeNG package is the lack of generic types/interfaces. Most of its event types have properties with "any" type value (which is not recommended). To fix that I have to manually create another generic interface by overriding the value I want to pass as a type parameter.

This is an example of how you are working interfaces without generics:
````typescript
export interface DropdownChangeEvent {
    /**
     * Browser event.
     */
    originalEvent: Event;
    /**
     * Selected option value
     */
    value: any;
}
````
And this is how it should be declared (Using "ValueT = any" will not break your current code that uses the interface):
````typescript
export interface DropdownChangeEvent<ValueT = any> {
    /**
     * Browser event.
     */
    originalEvent: Event;
    /**
     * Selected option value
     */
    value: ValueT;
}
````

With this solution, you are making a lot of developers save their time by not making extra types. Please do it for EVERY type/interface that has the property {value: any}

**Comment by BeGj** [comment]
Just make a merge request? 😅
